### PR TITLE
Fix redis installation on macos and amazonlinux

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
* Fix for https://github.com/aws/glide-for-redis/issues/781
Use shared GHA step to install redis.
* Activate amazon linux runs on our fork. How it works?

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -85,7 +84,7 @@ jobs:
               run: make test
 
     build-amazonlinux-latest:
-        if: github.repository_owner == 'aws'
+#        if: github.repository_owner == 'aws'
```
Comment (jonathanl-bq) [inline .github/workflows/go.yml:87]: Why was this required? Also, if we want to check this in, we probably shouldn't have commented out code.

File: .github/workflows/java.yml [hunk 2]
```diff
@@ -96,7 +94,7 @@ jobs:
                       benchmarks/results/**
 
     build-amazonlinux-latest:
-        if: github.repository_owner == 'aws'
+#        if: github.repository_owner == 'aws'
```
Comment (jonathanl-bq) [inline .github/workflows/java.yml:97]: Same here. Commented out code.

File: .github/workflows/go.yml (hunk 1 above)
Comment (Yury-Fridlyand) [reply .github/workflows/go.yml:87]: We didn't run CI on amazonlinux, because we dont have that runner on our repo. But flow worked, so I commented out that condition and fixed CI.
Not sure whether GHA uses amazonlinux or ubuntu there...
Waiting for a response from amazon team

Comment (Yury-Fridlyand) [reply .github/workflows/go.yml:87]: I want to keep this as is until all CI run on upstream. I'll remove if it passes or revert these files if it fails.

Comment (jonathanl-bq) [reply .github/workflows/go.yml:87]: I suppose that's okay. I think you could also make a draft PR upstream and test CI that way, but I don't care as long as it gets tested upstream eventually.

File: .github/workflows/node.yml [hunk 3]
```diff
@@ -161,19 +173,27 @@ jobs:
...
-              working-directory: ./node
-              run: |
-                  ln -s /usr/bin/redis6-server /usr/bin/redis-server
-                  ln -s /usr/bin/redis6-cli /usr/bin/redis-cli
-
             - name: Test compatibility
-              run: npm test -- -t "set and get flow works"
               working-directory: ./node
+              run: |
+                npm install --save-dev jest
```
Comment (jonathanl-bq) [inline .github/workflows/node.yml:198]: This change seems unrelated to the Redis installation. What is this for?

**Review by jonathanl-bq — APPROVED** [review]
One small question, but otherwise seems fine to me.

File: .github/workflows/node.yml (hunk 3 above)
Comment (Yury-Fridlyand) [reply .github/workflows/node.yml:198]: I found CI failed because of missing `jest`. I have no idea how it works on upstream, unfortunately.

Comment (jonathanl-bq) [reply .github/workflows/node.yml:198]: Maybe worth trying without `jest` on upstream in a draft and adding it in the real PR if still needed. Anyway, not a blocker.

Comment (Yury-Fridlyand) [reply .github/workflows/node.yml:198]: Sure, thanks
